packages/office-ui-fabric-react/src/components/Facepile/examples/Facepile.AddFace.Example.tsx

```diff
@@ -1,41 +1,32 @@
 import * as React from 'react';
-import { Facepile, IFacepileProps, OverflowButtonType } from 'office-ui-fabric-react/lib/Facepile';
+import { Facepile, OverflowButtonType } from 'office-ui-fabric-react/lib/Facepile';
 import { facepilePersonas } from '@uifabric/example-data';
 
-export interface IFacepileAddFaceExampleState {
-  numberOfFaces: number;
-}
+const overflowButtonProps = {
+  ariaLabel: 'More users',
+  onClick: (ev: React.MouseEvent<HTMLButtonElement>) => alert('overflow icon clicked'),
+};
 
-export class FacepileAddFaceExample extends React.Component<{}, IFacepileAddFaceExampleState> {
-  constructor(props: {}) {
-    super(props);
+export const FacepileAddFaceExample: React.FunctionComponent = () => {
+  const [numberOfFaces, setNumberOfFaces] = React.useState(0);
 
-    this.state = {
-      numberOfFaces: 0,
-    };
-  }
+  const addButtonProps = React.useMemo(
+    () => ({
+      ariaLabel: 'Add a new person',
+      onClick: (ev: React.MouseEvent<HTMLButtonElement>) => setNumberOfFaces(numberOfFaces + 1),
+    }),
+    [numberOfFaces],
+  );
 
-  public render(): JSX.Element {
-    const { numberOfFaces } = this.state;
-    const facepileProps: IFacepileProps = {
-      personas: facepilePersonas.slice(0, numberOfFaces),
-      maxDisplayablePersonas: 5,
-      overflowButtonProps: {
-        ariaLabel: 'More users',
-        onClick: (ev: React.MouseEvent<HTMLButtonElement>) => alert('overflow icon clicked'),
-      },
-      overflowButtonType: OverflowButtonType.descriptive,
-      showAddButton: true,
-      addButtonProps: {
-        ariaLabel: 'Add a new person',
-        onClick: (ev: React.MouseEvent<HTMLButtonElement>) =>
-          this.setState({
-            numberOfFaces: this.state.numberOfFaces + 1,
-          }),
-      },
-      ariaDescription: 'To move through the items use left and right arrow keys.',
-    };
-
-    return <Facepile {...facepileProps} />;
-  }
-}
+  return (
+    <Facepile
+      personas={facepilePersonas.slice(0, numberOfFaces)}
```

Comment:
This and `overflowPersonas` should potentially be memoized because otherwise it's creating a new array each time (same in other examples). Not sure if that causes any significant recalculations in the facepile case specifically, but it can definitely be problematic for some components.